apps/smart-contracts/core/test/unit/DepositHook.test.ts

```diff
@@ -201,6 +201,29 @@ describe('=> DepositHook', () => {
       expect(tokenSender.send).not.called
     })
 
+    it("doesn't call tokenSender.send() if fee > 0 but reward = 0", async () => {
+      await depositHook.connect(deployer).setAmountMultiplier(0)
+      expect(TEST_AMOUNT_BEFORE_FEE.sub(TEST_AMOUNT_AFTER_FEE)).gt(0)
+
+      await depositHook
+        .connect(collateral.wallet)
+        .hook(user.address, user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
+
+      expect(tokenSender.send).not.called
+    })
+
+    it("doesn't call tokenSender.send() if fee > 0 and reward > 0 but tokenSender not set", async () => {
```

Comment:
```suggestion
    it("doesn't call tokenSender.send() if tokenSender not set", async () => {
```

same for other hooks